### ISSUE TITLE
Pattern = (equals sign) gets interpreted as an assignment

### ISSUE THREAD

**Issue by dcasella** [koalaman/shellcheck#1412]
I didn't know whether to file this issue as a bug or feature request, so here's a "bug":

#### For bugs
- Rule Id: SC2034, SC1068
- My shellcheck version: online
- [x] I read the issue's wiki page
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env sh

grep = file.txt
```

#### Here's what shellcheck currently says:

```
Line 3:
grep = file.txt
^-- SC2034: grep appears unused. Verify use (or export if used externally).
     ^-- SC1068: Don't put spaces around the = in assignments.
```

#### Here's what I wanted or expected to see:

```
Line 3:
grep = file.txt
     ^-- SC2243: Add quotes around pattern string.
```

or no warn/err.

**Comment by koalaman** [comment]
I don't know how you could recognize this case more generally to give a specific warning, but I've updated the message to follow the parenthesized alternative convention that SC2034 already uses:

    SC1068: Don't put spaces around the = in assignments (or quote to make it literal).

Thanks!
